neps/nep-0621.md

```diff
@@ -0,0 +1,490 @@
+
+By standardizing the vault interface, this NEP aims to improve interoperability, reduce integration costs, and encourage consistent, secure practices for vault implementation across the NEAR ecosystem.
+
+## Motivation
+
+Vault contracts are a fundamental building block in modern DeFi, enabling users to pool assets for yield generation, liquidity provision, or other strategies while receiving tokenized shares that represent their proportional ownership. However, without a standardized interface, each vault implementation on NEAR may expose different method names, return formats, and accounting mechanisms, creating unnecessary friction for developers, integrators, and auditors.
+
+A consistent vault standard, inspired by ERC-4626, would provide multiple benefits:
+
+-   Interoperability – Wallets, DEXs, lending protocols, and other DeFi applications can integrate with any compliant vault without custom logic for each implementation.
+
+-   Reduced Integration Costs – Developers and projects save time and resources by building once against the standard interface rather than creating one-off integrations.
+
+-   Ecosystem Growth – Standardized vaults make it easier for new projects to leverage existing liquidity and composability, accelerating adoption across the NEAR DeFi ecosystem.
+
+By introducing this NEP, we aim to align vault design on NEAR with proven best practices from other blockchain ecosystems while optimizing for the unique features and requirements of NEP-141 fungible tokens.
+
+## Specification
+
+### Contract Interface
+
+The contract should implement the VaultCore trait.
+
+```rust
+/// Specification for a fungible token vault that issues NEP-141 compliant shares.
+///
+/// A FungibleTokenVault accepts deposits of an underlying NEP-141 compliant asset
+/// and issues NEP-141 compliant "shares" in return. These shares can be transferred
+/// and traded like any other NEP-141 token.
+///
+/// This trait extends:
+/// - [`FungibleTokenCore`] to provide NEP-141 functionality for shares.
+/// - [`FungibleTokenReceiver`] to receiver the underlying NEP-141 assets
+pub trait VaultCore:
+    FungibleTokenCore + FungibleTokenReceiver
+{
+    // ----------------------------
+    // Asset Information
+    // ----------------------------
+
+    /// Returns the [`AccountId`] of the underlying asset token contract.
+    ///
+    /// The asset **must** be NEP-141 compliant.
+    /// Implementations should store this as an immutable configuration value.
+    fn asset(&self) -> AccountId;
+
+    /// Returns the total amount of underlying assets represented by all shares in existence.
+    ///
+    /// **Important:**
+    /// - Represents the vault's *total managed value*, not just assets held in the contract.
```

Comment:
I understand where it comes from, but two "asset"/"assets" terms create a wild confusion. "Asset" means FT address, and "assets" means "amount of that one underlying asset (FT token)". Perhaps, it can be better addressed, e.g. "assets" could be called as "underlying asset's amount" in the docs comments.